### PR TITLE
layer-shell: use output_damage_{whole,from}_local_surface

### PR DESCRIPTION
**Pull request by agx**
Reuse what already handles rotation and scaling. This unbreaks popups
on rotated or scaled outputs.

Can be tested with layer-shell example.

### CODE REVIEW

File: rootston/layer_shell.c [hunk 1]
```diff
@@ -297,35 +297,32 @@ static void popup_handle_map(struct wl_listener *listener, void *data) {
 	struct roots_layer_surface *layer = popup->parent;
 	struct wlr_output *wlr_output = layer->layer_surface->output;
 	struct roots_output *output = wlr_output->data;
-	struct wlr_box geom;
-	memcpy(&geom, &popup->wlr_popup->geometry, sizeof(struct wlr_box));
-	geom.x += layer->geo.x;
-	geom.y += layer->geo.y;
-	wlr_output_damage_add_box(output->damage, &geom);
+	int ox = popup->wlr_popup->geometry.x;
```
Comment (emersion) [inline rootston/layer_shell.c]: Can you sum `layer->geo.x` in the `ox` definition? Otherwise it's not really the position in output-local coords.

**Comment by agx** [comment]
@emersion yeah that much less confusing. Updated.

File: rootston/layer_shell.c [hunk 2]
```diff
@@ -297,35 +297,32 @@ static void popup_handle_map(struct wl_listener *listener, void *data) {
...
 	struct roots_output *output = wlr_output->data;
-	struct wlr_box geom;
-	memcpy(&geom, &popup->wlr_popup->geometry, sizeof(struct wlr_box));
-	geom.x += layer->geo.x;
-	geom.y += layer->geo.y;
-	wlr_output_damage_add_box(output->damage, &geom);
+	int ox = popup->wlr_popup->geometry.x + layer->geo.x;
+	int oy = popup->wlr_popup->geometry.y + layer->geo.y;
+	output_damage_from_local_surface(output, popup->wlr_popup->base->surface,
+		ox, oy, 0);
```
Comment (emersion) [inline rootston/layer_shell.c]: I think this should be `output_damage_whole_local_surface`

**Comment by agx** [comment]
BTW I wonder if layer_shell::unmap shouldn't better use output_damage_whole_local_surface as well?

File: rootston/layer_shell.c (hunk 2 above)
Comment (emersion) [reply rootston/layer_shell.c]: Basically, `from` can only be used in commit handlers

**Comment by emersion** [comment]
>BTW I wonder if layer_shell::unmap shouldn't better use output_damage_whole_local_surface as well?

Probably. Same for `map`. Patches welcome!

**Comment by agx** [comment]
@emersion  Ah, o.k. I probably didn't notice since the commit came right after the map. Thanks!

**Comment by agx** [comment]
Fixed layer_shell::{,un}map too

**Pull request merged**